src/components/structures/RoomView.js

```diff
@@ -414,6 +420,7 @@ module.exports = React.createClass({
             this.setState({
                 room: room
             });
+            this._initTimeline(this.props);
```

Comment:
Good point: I was confused by the fact that we never listened for the Room event, but I think this is the event we actually want so I've added a listener and separated it.